docusaurus/docs/extensions/advanced/create-page-in-an-existing-product.md

```diff
@@ -0,0 +1,171 @@
+# Create a page in an existing product
+
+Given that you have covered all of the topics in [Extensions API](../api/overview.md), especially the [Navigation & Pages](../api/nav/products.md) area, we will show an example of how you can create an extension that adds a new page to an existing product `Fleet`. This doesn't need any product registration.
+
+Let's take into account this simple `index.ts` entry for an extension:
+
+```ts
+// ./index.ts
+import { importTypes } from '@rancher/auto-import';
+import { IPlugin } from '@shell/core/types';
+import extensionRouting from './routing/extension-routing';
+
+// Init the package
+export default function(plugin: IPlugin) {
+  // Auto-import model, detail, edit from the folders
+  importTypes(plugin);
+
+  // Provide extension metadata from package.json
+  // it will grab information such as `name` and `description`
+  plugin.metadata = require('./package.json');
+
+  // Load a product
+  plugin.addProduct(require('./product'));
+
+  // Add Vue Routes
+  plugin.addRoutes(extensionRouting);
+}
+```
+
+The `product.ts` config will then define the "pages/views" we want to add taking into account the route structure of a `top-level product` such as `Fleet`:
```

Comment:
I think it would be helpful to clarify to users that product.ts here is expanding on the product definition in shell/config/product/fleet. It may be confusing that this page begins with "This doesn't need any product registration." then has a file called product.ts that needs to be 'registered' in index.ts